src/common/UserAvatar.tsx

```diff
@@ -1,14 +1,16 @@
-import { Avatar } from '@mui/material'
+import { Avatar, AvatarProps } from '@mui/material'
 import jdenticon from 'jdenticon/standalone'
 import { useMemo } from 'react'
 
-import { useMetamaskZkpSnapContext } from '@/hooks'
+interface UserAvatarProps extends AvatarProps {
+  userDid: string
+  size?: number
+}
 
-const UserAvatar = ({ ...rest }) => {
-  const { userDid } = useMetamaskZkpSnapContext()
+const UserAvatar = ({ userDid, size = 48, ...rest }: UserAvatarProps) => {
   const userAvatar = useMemo(() => {
-    return jdenticon.toSvg(userDid, 48)
-  }, [userDid])
+    return jdenticon.toSvg(userDid, Number(size))
```

Comment:
already a number, no conversion is needed

Reply:
now it is converted with "spacing" which returns string, but the function accepts type number